lib/index.js

```diff
@@ -99,6 +99,12 @@ exports.download = (opts, url, reporter) => {
         });
         return;
       }
+      if (res.request.uri.pathname.indexOf('photo-1446704477871-62a4972035cd') >= 0) {
+        reject({
+          message: 'The response is the image: we couldn\'t find that photo.'
+        });
+        return;
+      }
```

Comment:
This seems as good as we can probably make it. If they change the 404 image, the worst that can happen is this code doesn't run.